private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -61,39 +206,77 @@ export class TemplateServiceClient {
         const newTemplate: ITemplate = {
             instances: [templateInstance],
             tags: [],
-            owner: owner,
-            isPublished: false
+            owner: this.ownerID!,
+            isPublished: isPublished
         }
 
-        return TemplateServiceClient.storageProvider.insertTemplate(newTemplate);
+        return this.storageProvider.insertTemplate(newTemplate);
     }
 
     /**
      * @public
      * Get entry point. 
+     * Returns specified templates
      * @param {string} templateId - unique template id
      * @param {boolean} isPublished 
      * @param {string} templateName - name to query for
      * @param {string} version - version number
+     * @param {boolean} owned - If false, will retrieve all public templates regardless of owner
      * @returns Promise as valid json 
      */
-    public async getTemplates(templateId?: string, isPublished?: boolean, templateName?: string, version?: number): Promise<JSONResponse<ITemplate[]>> {
-        let owner = TemplateServiceClient.authProvider.getOwner();
+    public async getTemplates(templateId?: string, isPublished?: boolean, templateName?: string, version?: number, owned?: boolean): Promise<JSONResponse<ITemplate[]>> {
```

Comment:
templateId here is optional? what does it mean this to be optional? Do you need to specifically consider this case?

Reply:
I added getTemplates to either get a template based on the id, or if an id was not passed (id is optional), the function can also retrieve all of the user's owner templates or published templates that are not owner by the user